### PR TITLE
New losses

### PR DESCRIPTION
**Pull request by csudre**
## Status
**READY**

## Description
A few sentences describing the overall goals of the pull request's commits.
This proposes new losses to deal with multi rater consistency, volume enforcement and soft l1

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X ] My code follows the code style of this project.
- [ X] I have added tests to cover my changes.
- [X ] All new and existing tests passed.

## Todos
- [ ] Tests
- [ ] Documentation

## Impacted Areas in Application
List general components of the application that this PR will affect:
* Added losses

### CODE REVIEW

File: niftynet/layer/loss_classification_multi.py [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+                            pred, pred_mul))
+
+            if weight_batch is not None:
+                return tf.reduce_mean(weight_batch/tf.reduce_sum(
+                    weight_batch) * data_loss[0])
+            else:
+                return tf.reduce_mean(data_loss)
+
+
+def labels_to_one_hot(ground_truth, num_classes=1, nrater=6):
```
Comment (wyli) [inline niftynet/layer/loss_classification_multi.py]: `nrater` is not used in this function, perhaps we could `from niftynet.layer.loss_segmentation import labels_to_one_hot` instead of creating a new copy here?

File: niftynet/layer/loss_classification_multi.py [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+                 ground_truth=None,
+                 weight_batch=None,
+                 var_scope=None, ):
+        """
+        Compute loss from `prediction` and `ground truth`,
+
+        if `prediction `is list of tensors, each element of the list
+        will be compared against `ground_truth`.
+
+        :param prediction: input will be reshaped into (N, num_classes)
```
Comment (wyli) [inline niftynet/layer/loss_classification_multi.py]: add a few lines of doc for the expected shape of param `pred_multi`and `pred_ave`?

File: niftynet/layer/loss_classification_multi.py (hunk 1 above)
Comment (csudre) [reply niftynet/layer/loss_classification_multi.py]: The second output (output_shape) is required and not present in the function of segmentation loss. For now I have left but am happy to change loss_segmentation instead and clall it from there

File: niftynet/layer/loss_classification_multi.py (hunk 2 above)
Comment (csudre) [reply niftynet/layer/loss_classification_multi.py]: Done!

**Review by Zach-ER — COMMENTED** [review]
Looks good. I would add comments in niftynet/layer/loss_classification_multi on lines 77 to explain the many cases in these nested statements, because it is quite dense to read.
